x509/src/general_name.rs

```diff
@@ -0,0 +1,417 @@
+/// GeneralNames as defined in [RFC 5280 Section 4.2.1.6].
+use crate::alloc::string::ToString;
+use crate::Name;
+use der::asn1::{Any, ContextSpecific, Ia5String, ObjectIdentifier, OctetString};
+use der::{Decodable, DecodeValue, Decoder, Length, Sequence, TagMode, TagNumber};
+
+/// OtherName as defined in [RFC 5280 Section 4.2.1.6] in support of the Subject Alternative Name extension.
+///
+/// ```text
+///    OtherName ::= SEQUENCE {
+///         type-id    OBJECT IDENTIFIER,
+///         value      [0] EXPLICIT ANY DEFINED BY type-id }
+/// ```
+///
+/// [RFC 5280 Section 4.2.1.6]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct OtherName<'a> {
+    /// type-id    OBJECT IDENTIFIER,
+    pub type_id: ObjectIdentifier,
+
+    /// value      [0] EXPLICIT ANY DEFINED BY type-id }
+    pub value: Any<'a>,
+}
+
+impl<'a> DecodeValue<'a> for OtherName<'a> {
+    fn decode_value(decoder: &mut Decoder<'a>, _length: Length) -> der::Result<Self> {
+        let type_id = decoder.decode()?;
+        let value = decoder.decode()?;
+        Ok(Self { type_id, value })
+    }
+}
+
+impl<'a> Sequence<'a> for OtherName<'a> {
+    fn fields<F, T>(&self, f: F) -> ::der::Result<T>
+    where
+        F: FnOnce(&[&dyn der::Encodable]) -> ::der::Result<T>,
+    {
+        f(&[&self.type_id, &self.value])
+    }
+}
+
+/// GeneralNames as defined in [RFC 5280 Section 4.2.1.6] in support of the Subject Alternative Name extension.
+///
+/// ```text
+/// GeneralNames ::= SEQUENCE SIZE (1..MAX) OF GeneralName
+/// ```
+///
+/// [RFC 5280 Section 4.2.1.6]: https://datatracker.ietf.org/doc/html/rfc5280#section-4.2.1.6
+pub type GeneralNames<'a> = alloc::vec::Vec<GeneralName<'a>>;
```

Comment:
I'd suggest adding a `use` directive for `alloc::vec::Vec` at the top of the file

Reply:
Fixed.